Makefile

```diff
@@ -1,29 +1,41 @@
-VENV=venv
-PIP=$(VENV)/bin/pip
-PYTHON=$(VENV)/bin/python
+VENV := venv
+PYTHON := python
+BIN = $(VENV)/bin
+PYTHON_VERSION = $(shell $(PYTHON) -c 'import sys;print(sys.version_info[0])')
 
 init:
-	test `command -v python3` || echo Please install python3
-	[ -d $(VENV) ] || python3 -m venv $(VENV)
-	$(PIP) install -r requirements_dev.txt
-	$(PIP) install -e .
+	test `command -v $(PYTHON)` || echo Please install $(PYTHON)
+	pip install virtualenv
+	[ -d $(VENV) ] || virtualenv -p $(PYTHON) $(VENV)
+ifeq ($(PYTHON_VERSION),3)
+	$(BIN)/pip install -r requirements_dev.txt
+	$(BIN)/pre-commit install
+else ifeq ($(PYTHON_VERSION),2)
+	$(BIN)/pip install -r requirements_dev2.txt
+endif
+	$(BIN)/pip install -e .
 
 lint:
-	$(PYTHON) -m black -q quadbin tests setup.py
-	$(PYTHON) -m flake8 quadbin tests setup.py
+ifeq ($(PYTHON_VERSION),3)
+	$(BIN)/black -q quadbin/ tests/ setup.py
+endif
+	$(BIN)/flake8 quadbin/ tests/ setup.py
 
 test:
-	$(PYTHON) -m pytest tests
+	$(BIN)/pytest --cov=quadbin tests/unit/
+
+test-benchmark:
+	$(BIN)/pytest tests/benchmark/ --benchmark-histogram
 
 publish-pypi:
 	rm -rf ./dist/*
-	$(PYTHON) setup.py sdist bdist_wheel
-	$(PYTHON) -m twine upload dist/*
+	$(BIN)/python setup.py sdist bdist_wheel
+	$(BIN)/twine upload dist/*
 
 publish-test-pypi:
 	rm -rf ./dist/*
-	$(PYTHON) setup.py sdist bdist_wheel
-	$(PYTHON) -m twine upload --repository-url https://test.pypi.org/legacy/ dist/* --verbose
+	$(BIN)/python setup.py sdist bdist_wheel
+	$(BIN)/twine upload --repository-url https://test.pypi.org/legacy/ dist/* --verbose
```

Comment:
As I understand from this, we publish the module by using python 2 right? That's why it's the only that needs wheel

Reply:
The packages can be published using Python 3, even if they are available in Python 2 too.